src/treebuilders/MapCalculator.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * MRCPP, a numerical library based on multiresolution analysis and
+ * the multiwavelet basis which provide low-scaling algorithms as well as
+ * rigorous error control in numerical computations.
+ * Copyright (C) 2020 Stig Rune Jensen, Jonas Juselius, Luca Frediani and contributors.
+ *
+ * This file is part of MRCPP.
+ *
+ * MRCPP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MRCPP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with MRCPP.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * For information on the complete list of contributors to MRCPP, see:
+ * <https://mrcpp.readthedocs.io/>
+ */
+
+#pragma once
+
+#include "TreeCalculator.h"
+
+namespace mrcpp {
+
+template <int D> class MapCalculator final : public TreeCalculator<D> {
+public:
+    MapCalculator(std::function<double(double)> fm, FunctionTree<D> &inp)
+            : func(&inp)
+            , fmap(fm) {}
```

Comment:
It might be slightly more efficient to do:
```c++
fmap(std::move(fm))
```

Reply:
OK. But I doubt efficiency plays a role at this level. This is called once per function.